pkg/config/config.go

```diff
@@ -814,6 +815,56 @@ func sanitizeAPIKey(config Config) {
 	config.Set("api_key", strings.TrimSpace(config.GetString("api_key")))
 }
 
+//trims any forward slashes from the end of various config URL's (site, dd_url, and additional endpoints)
+func trimTrailingSlashFromURLS(config Config) error {
+	var urls = []string{
+		"site",
+		"dd_url",
+	}
+	var additionalEndpointSelectors = []string{
+		"additional_endpoints",
+		"apm_config.additional_endpoints",
+		"logs_config.additional_endpoints",
+		"process_config.additional_endpoints",
+	}
+
+	for _, domain := range urls {
+		key := config.GetString(domain)
+		if len(key) == 0 {
+			continue
+		}
+		for {
+			if key[len(key)-1:] != "/" {
+				break
+			}
+			key = strings.TrimSuffix(key, "/")
+		}
+		config.Set(domain, key)
+	}
+
+	for _, es := range additionalEndpointSelectors {
+		additionalEndpoints := make(map[string][]string)
+		sanitizedAdditionalEndpoints := make(map[string][]string)
+
+		err := config.UnmarshalKey(es, &additionalEndpoints)
+
+		if err != nil {
+			return err
+		}
+		for domain, key := range additionalEndpoints {
+			for {
+				domain = strings.TrimSuffix(domain, "/")
+				if domain[len(domain)-1:] != "/" {
+					sanitizedAdditionalEndpoints[domain] = key
+					break
+				}
```

Comment:
This one may failed if domain is equal to "", it should not append in normal case, but I think we should handle it. You can handle it the same way you did above, as it does not change the runtime behaviour.